bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,207 @@
+          # delete the old key to be sure it is not there
+          kubectl delete --ignore-not-found=true csr/$(hostname)
+
+          cat <<EOF | kubectl apply -f -
+          apiVersion: certificates.k8s.io/v1
+          kind: CertificateSigningRequest
+          metadata:
+            name: $(hostname)
+          spec:
+            request: ${csr_64}
+            signerName: network.openshift.io/signer
+            usages:
+            - ipsec tunnel
+          EOF
+
+          until [ ! -z $(kubectl get csr/$(hostname) -o jsonpath='{.status.certificate}' 2>/dev/null) ]
+          do
+            ((counter++))
+            if [ $counter -gt 60 ];
+            then
+                    echo "Unable to sign certificate after $counter seconds"
+                    exit 1
+            fi
+          done
+
+          kubectl get csr/$(hostname) -o jsonpath='{.status.certificate}' | base64 -d | openssl x509 -outform pem -text -out ipsec-cert.pem
+
+          kubectl delete csr/$(hostname)
+
+          cat /signer-ca/tls.crt | openssl x509 -outform pem -text > /etc/openvswitch/keys/ipsec-cacert.pem
+
+          ovs-vsctl --retry -t 60 set Open_vSwitch . other_config:certificate=/etc/openvswitch/keys/ipsec-cert.pem \
+                                                     other_config:private_key=/etc/openvswitch/keys/ipsec-privkey.pem \
+                                                     other_config:ca_cert=/etc/openvswitch/keys/ipsec-cacert.pem
+
+          popd
+
+          # Workaround for https://github.com/libreswan/libreswan/issues/373
+          ulimit -n 1024
+
+          /usr/libexec/ipsec/addconn --config /etc/ipsec.conf --checkconfig
+          # Check for kernel modules
+          /usr/libexec/ipsec/_stackmanager start
+          # Check for nss database status and migration
+          /usr/sbin/ipsec --checknss
+          # Check for nflog setup
+          /usr/sbin/ipsec --checknflog
+          # Start the actual IKE daemon
+          /usr/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf --logfile /var/log/openvswitch/libreswan.log
+
+          # Workaround for https://mail.openvswitch.org/pipermail/ovs-dev/2020-October/375734.html
+          OVS_LOGDIR=/var/log/openvswitch OVS_RUNDIR=/var/run/openvswitch OVS_PKGDATADIR=/usr/share/openvswitch /usr/share/openvswitch/scripts/ovs-ctl --ike-daemon=libreswan start-ovs-ipsec
+
+          # Wait for the nbdb container in the ovnkube-master pod to become ready before we try to write to
+          # it.
+          if kubectl wait --for=condition=Ready pod -l app=ovnkube-master -n ${ovn_kubernetes_namespace} --timeout 600s; then
+            echo "ovnkube-master is not Ready, unable to write to OVNNB DB"
+            exit 2
+          else
+            ovn-nbctl set nb_global . ipsec=true
+          fi
+
+          while true; do
+            sleep 10
+          done
+        env:
+        - name: OVS_LOG_LEVEL
+          value: info
+        - name: K8S_NODE
```

Comment:
You don't seem to be using this environment variable anywhere, but instead use `hostname`, so this could be removed right?

Reply:
Yes, you are right - I copied this spec file from the OVS one. I will do another parse of it to make sure I havent missed anything else.

Reply:
We use it for overriding loglevels on a per-node basis - perhaps that mechanism is useful here?

Check out the env-overrides bit in the node daemonset.

Reply:
Ok, based on Casey's comment, I will keep this.